options.go

```diff
@@ -51,6 +54,7 @@ type dialerConfig struct {
 	authCredentials          *auth.Credentials
 	iamLoginTokenProvider    auth.TokenProvider
 	useragents               []string
+	applicationName          string
```

Comment:
`applicationID` with a TODO to set this by default using GCE VM Name, GKE pod name, Serverless app instance name (whatever that is).